ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionQueryBuilder.java

```diff
@@ -203,19 +214,23 @@ CompactionQueryBuilder setIsDeleteDelta(boolean deleteDelta) {
    * @param resultTableName the name of the table we are running the operation on
    * @throws IllegalArgumentException if compactionType is null
    */
-  CompactionQueryBuilder(CompactionType compactionType, Operation operation,
+  CompactionQueryBuilder(CompactionType compactionType, Operation operation, boolean crud,
```

Comment:
I don't really like passing an extra param (crud) that could be determined from the compactionType.